app/drivers/gpio/max7318/gpio_max7318.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT maxim_max7318
+
+/**
+ * @file Driver for MAX7318 I2C-based GPIO driver.
+ */
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <sys/byteorder.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <drivers/ext_power.h>
+
+#include "gpio_max7318.h"
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(gpio_max7318);
+
+/**
+ * @brief Read the value of two consecutive registers
+ *
+ * Read two consecutive bytes from the register at address `reg` and `reg + 1`,
+ * typically reading from registers for port 0 and 1 simultaneously.
+ *
+ * @param dev   The max7318 device.
+ * @param reg   Register to read (the PORT0 of the pair of registers).
+ * @param buf   Buffer to read data into.
+ *
+ * @return 0 if successful, failed otherwise.
+ */
+static int read_registers(const struct device *dev, uint8_t reg, uint16_t *buf) {
+    struct max7318_drv_data *const drv_data = (struct max7318_drv_data *const)dev->data;
+    uint16_t dev_addr = ((struct max7318_config *)dev->config)->device_addr;
+
+    uint16_t data = 0;
+    int ret = i2c_burst_read(drv_data->i2c, dev_addr, reg, (uint8_t *)&data, sizeof(data));
```

Comment:
Ditto, move to the `_dt` version of this.

Reply:
Done.